service/lib/agama/dbus/storage/volume_conversion/from_dbus.rb

```diff
@@ -0,0 +1,147 @@
+# frozen_string_literal: true
+
+# Copyright (c) [2023] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "agama/storage/volume"
+require "agama/storage/volume_templates_builder"
+require "y2storage/disk_size"
+require "y2storage/filesystems/type"
+
+module Agama
+  module DBus
+    module Storage
+      module VolumeConversion
+        # Volume conversion from D-Bus format.
+        class FromDBus
+          # @param dbus_volume [Hash]
+          # @param config [Agama::Config]
+          def initialize(dbus_volume, config:)
+            @dbus_volume = dbus_volume
+            @config = config
+          end
+
+          # Performs the conversion from D-Bus format.
+          #
+          # @return [Agama::Storage::Volume]
+          def convert
+            builder = Agama::Storage::VolumeTemplatesBuilder.new_from_config(config)
+            volume = builder.for(dbus_volume["MountPath"] || "")
+
+            volume.tap do |target|
+              dbus_volume.each do |dbus_property, dbus_value|
+                converter = CONVERSIONS[dbus_property]
+                # FIXME: likely ignoring the wrong attribute is not the best
```

Comment:
same here. at least log it. Exception and proper error report would be even better to quickly catch it.